ethereum/contracts/Scale.sol

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.6.2;
+
+contract Scale {
+
+    // Read a byte at a specific index and return it as type uint8
+    function readByteAtIndex(bytes memory data, uint8 index)
+        internal
+        pure
+        returns (uint8)
+    {
+        return uint8(data[index]);
+    }
+
+    // Reverse uint8[] array in place
+    function reverse(uint8[] memory arr)
+        internal
+        pure
+        returns(uint8[] memory)
+    {
+        for (uint i = 0; i < arr.length/2; i++) {
+            uint8 current = arr[i];
+            uint256 otherIndex = arr.length - i - 1;
+            arr[i] = arr[otherIndex];
+            arr[otherIndex] = current;
+        }
+        return arr;
+    }
+
+    // Convert an uint8[] array to uint256
+    function uint8ArrayToUint256(uint8[] memory arr)
+        internal
+        pure
+        returns (uint256)
+    {
+        uint256 number;
+        for(uint i = 0; i < arr.length; i++){
+            number = number+uint(arr[i])*(2**(8*(arr.length-(i+1))));
+        }
+        return number;
+    }
+
+      // Convert bytes (big endian) to little endian format
+    function bytesToLittleEndian(bytes memory arr)
+        internal
+        pure
+        returns (uint256)
+    {
+        uint256 number;
+        for(uint i = 0; i < arr.length; i++){
+            number = number+uint(uint8(arr[i]))*(2**(8*(arr.length-(i+1))));
+        }
+        return number;
+    }
+
+     // Reverse an array of bytes in place
+    function reverseBytes(bytes memory arr)
+        internal
+        pure
+        returns(bytes memory)
+    {
+        for (uint i = 0; i < arr.length/2; i++) {
+            bytes1 current = arr[i];
+            uint256 otherIndex = arr.length - i - 1;
+            arr[i] = arr[otherIndex];
+            arr[otherIndex] = current;
+        }
+        return arr;
+    }
+
+    function decodeUint256(bytes memory data)
+        public
+        pure
+        returns (uint256)
+    {
+        bytes memory reversed = reverseBytes(data);
+        uint256 lEndian = bytesToLittleEndian(reversed);
+        return lEndian;
+    }
+
+    // Decodes a SCALE encoded compact unsigned integer
+    function decodeUintCompact(bytes memory data)
+        public
+        pure
+        returns (uint256)
+    {
+        uint8 b = readByteAtIndex(data, 0);           // read the first byte
+        uint8 mode = b & 3;                           // bitwise operation
+
+        if(mode == 0) {                               // [0, 63]
```

Comment:
According to the recommended style guide at https://solidity.readthedocs.io/en/v0.5.3/style-guide.html there should be a space between the `if` and the first parenthesis.

Reply:
Noted, fixed in https://github.com/Snowfork/polkadot-ethereum/pull/51/commits/d3d5a2bc7a79b6949357ec8082aae20ce6b9c1b6